packages/driver/src/util/serialization/log.ts

```diff
@@ -0,0 +1,391 @@
+  type?: any
+  id?: string
+  tagName: string
+  attributes: { [key: string]: string }
+  innerHTML: string
+  serializationKey: 'dom'
+}
+
+interface PreprocessedFunction {
+  value: any
+  serializationKey: 'function'
+}
+
+/**
+ * Takes an HTMLElement that might be a <body> for a given snapshot or any other element, likely pertaining to log consoleProps,
+ * on the page that needs to be preprocessed for serialization. The method here is to do a very shallow serialization,
+ * by trying to make the HTML as stateful as possible before preprocessing.
+ *
+ * @param {HTMLElement} props - an HTMLElement
+ * @returns {PreprocessedHTMLElement} a preprocessed element that can be fed through postMessage() that can be reified in the primary.
+ */
+const preprocessDomElement = (props: HTMLElement) => {
+  // hydrate values in HTML copy so when serialized they show up correctly in snapshot. This is important for input boxes with typing and other 'value' attributes
+  props.querySelectorAll('input').forEach((input: HTMLInputElement) => {
+    input.setAttribute('value', input.value)
+    if (input.checked) {
+      input.setAttribute('checked', `${input.checked}`)
+    }
+  })
+
+  // TODO: figure out reifying option selection
+  // props.querySelectorAll('option').forEach((option) => {
+  //   if (option.selected) {
+  //     option.setAttribute('selected', option.selected)
+  //   }
+  // })
+
+  const el: PreprocessedHTMLElement = {
+    value: (props as HTMLInputElement)?.value,
+    type: (props as HTMLInputElement)?.type,
+    id: props?.id,
+    tagName: props.tagName,
+    attributes: {},
+    innerHTML: props.innerHTML,
+    serializationKey: 'dom',
+  }
+
+  // get all attributes and classes off the element
+  props.getAttributeNames().forEach((attributeName) => {
+    el.attributes[attributeName] = props.getAttribute(attributeName) || ''
+  })
+
+  return el
+}
+
+/**
+ * Takes an HTMLElement that might be a <body> for a given snapshot or any other element, likely pertaining to log consoleProps,
+ * on the page that needs to be preprocessed for serialization. The method here is to do a very shallow serialization,
+ * by trying to make the HTML as stateful as possible before preprocessing.
+ *
+ * @param {PreprocessedHTMLElement} props - a preprocessed element that was fed through postMessage() that need to be reified in the primary.
+ * @returns {HTMLElement} a reified element that can be used in a snapshot body or consoleProps.
+ */
+const reifyDomElement = (props: any) => {
+  const reifiedEl = document.createElement(props.tagName)
+
+  if (props.value) {
+    reifiedEl.value = props.value
+  }
+
+  try {
+    if (props.type) {
+      reifiedEl.type = props.type
+    }
+  } catch (e) {
+    // swallow this. certain elements this is a read-only property on (such as <select>)
+  }
+
+  if (props.id) {
+    reifiedEl.id = props.id
+  }
+
+  reifiedEl.innerHTML = props.innerHTML
+
+  Object.keys(props.attributes).forEach((attribute) => {
```

Comment:
I am not sure it's clear here, but `class` is considered an attribute and is stored here in the same way it is presented in the element. So if n element is something like 
```js
<div class="class1 @class23.    large_spaces_between" data-cy="foobar" id="foobar"/>
```
 it is reified as 
```js
<div class="class1 @class23.    large_spaces_between" data-cy="foobar" id="foobar"/>
```
except order might not be preserved, so it could technically be something like this below, which should not matter for element selection
```js
<div id="foobar" data-cy="foobar"  class="class1 @class23.    large_spaces_between"/>
```